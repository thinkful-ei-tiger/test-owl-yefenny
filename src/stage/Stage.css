.stage {
    background-color: #686d76;
    flex: 2;
}
.partners{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
}
.personal-stage{
    display: flex;
    justify-content: center;
    margin: 10px 0;
}
.personal-img{
    display: flex;
    display: column;
    width: 150px;
    height: 90px;
    justify-content: center;
    background-color: #373a40;
    border-radius: 3%;
}
.personal-img img {
    align-self: center;
    width: 50px;
    height: 50px;
    border-radius: 50%;
}
