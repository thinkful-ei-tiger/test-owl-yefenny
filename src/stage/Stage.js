import React, { Component } from 'react';
import PARTICIPANTS from '../dummy';
import Partner from '../partner/Partner';
import './Stage.css';

export default class Stage extends Component {
  render() {
    const id = 1;
    const participants = PARTICIPANTS.filter(
      (participant) => participant.id !== id
    );
    const personal = PARTICIPANTS.find((participant) => participant.id === id);
    const partners = participants
      .filter((participant) => participant.onStage !== false)
      .map((participant) => {
        return (
          <Partner
            key={participant.id}
            name={participant.name}
            avatar={participant.avatar}
          />
        );
      });
    return (
      <div className='stage'>
        <div className='partners'>{partners}</div>
        <div className='personal-stage'>
          <div className='personal-img'>
            <img src={personal.avatar} alt='personal' />
          </div>
        </div>
      </div>
    );
  }
}
