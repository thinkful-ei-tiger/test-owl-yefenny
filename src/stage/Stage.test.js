import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './Stage';

describe('Stage', () => {
  it('renders without crushing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Stage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('UI renders without problem', () => {
    const tree = renderer.create(<Stage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
