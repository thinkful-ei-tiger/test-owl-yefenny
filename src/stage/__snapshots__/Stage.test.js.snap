// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage UI renders without problem 1`] = `
<div
  className="stage"
>
  <div
    className="partners"
  >
    <div
      className="partner"
    >
      <div
        className="header"
      >
        <h2>
          Caty Flucker
        </h2>
        <div
          className="icons"
        >
          <span>
            <img
              alt="mute"
              src="volume.svg"
            />
            <img
              alt="search"
              src="search.svg"
            />
          </span>
        </div>
      </div>
      <div
        className="avatar"
      >
        <img
          alt="profile"
          src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
        />
      </div>
    </div>
    <div
      className="partner"
    >
      <div
        className="header"
      >
        <h2>
          Axe Kubicka
        </h2>
        <div
          className="icons"
        >
          <span>
            <img
              alt="mute"
              src="volume.svg"
            />
            <img
              alt="search"
              src="search.svg"
            />
          </span>
        </div>
      </div>
      <div
        className="avatar"
      >
        <img
          alt="profile"
          src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
        />
      </div>
    </div>
    <div
      className="partner"
    >
      <div
        className="header"
      >
        <h2>
          Ashla Attwool
        </h2>
        <div
          className="icons"
        >
          <span>
            <img
              alt="mute"
              src="volume.svg"
            />
            <img
              alt="search"
              src="search.svg"
            />
          </span>
        </div>
      </div>
      <div
        className="avatar"
      >
        <img
          alt="profile"
          src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
        />
      </div>
    </div>
  </div>
  <div
    className="personal-stage"
  >
    <div
      className="personal-img"
    >
      <img
        alt="personal"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </div>
  </div>
</div>
`;
